File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.py


def solution(n, m):
    lcm = LCM(n, m)
    gcd = GCD(n, m)
    return [gcd,lcm]

def GCD(n, m):
    a = max(n,m)
    b = min(n,m)
    while (b>0):
        a,b = b, a%b
    return a

def LCM(n, m):
    gcd = GCD(n, m)
    return abs(n * m) // gcd
        

File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.js
function solution(n, m) {  
    let originM = m;
    let originN = n;
    
    // 최대 공배수
    while(m !== 0){
        let keep = m;
        m = n % m;
        n = keep;
    }
    const gcd = n;
    
    // 최소 공배수
    const lcm = (originN * originM) / gcd;
    
    return [gcd, lcm]
}